src/endpoint/s3/s3_utils.js

```diff
@@ -263,7 +263,7 @@ function set_response_object_md(res, object_md) {
     res.setHeader('ETag', '"' + object_md.etag + '"');
     res.setHeader('Last-Modified', time_utils.format_http_header_date(new Date(object_md.create_time)));
     res.setHeader('Content-Type', object_md.content_type);
-    res.setHeader('Content-Length', object_md.size);
+    res.setHeader('Content-Length', object_md.content_length === undefined ? object_md.size : object_md.content_length);
```

Comment:
Ok so now this logic moved here, but I still don't understand the case when the size property is not set the way we need, can you explain?

Reply:
Content-Length is not always the size of object. In the case of range read or read with partNumber, it is the length of range to be read.

Reply:
So you're saying that this calculation below is not working correct with partNumber? Can you explain? Maybe we need to adjust that code.
https://github.com/noobaa/noobaa-core/blob/abbafc463045736a4a2be973b1e15c282cb41d97/src/sdk/namespace_s3.js#L532-L537

Reply:
The format of Content-Range is like 'content-range': 'bytes 100200-280000/3145735'. Given the example, the line at 532 returns 3145735 which is the size of entire object.

Reply:
will defer to new issue.